lib/modules/uscore_v3.1.1/bodyheight_sequence.rb

```diff
@@ -286,33 +286,39 @@ def patient_ids
         patient_ids.each do |patient|
           next unless @observation_ary[patient].present?
 
-          search_params = {
-            'patient': patient,
-            'category': get_value_for_search_param(resolve_element_from_path(@observation_ary[patient], 'category') { |el| get_value_for_search_param(el).present? }),
-            'date': get_value_for_search_param(resolve_element_from_path(@observation_ary[patient], 'effective') { |el| get_value_for_search_param(el).present? })
-          }
+          observation_ary = Array.wrap(@observation_ary[patient])
 
-          next if search_params.any? { |_param, value| value.nil? }
+          observation_ary.each do |observation|
+            search_params = {
+              'patient': patient,
+              'category': get_value_for_search_param(resolve_element_from_path(observation, 'category') { |el| get_value_for_search_param(el).present? }),
+              'date': get_value_for_search_param(resolve_element_from_path(observation, 'effective') { |el| get_value_for_search_param(el).present? })
+            }
 
-          resolved_one = true
+            next if search_params.any? { |_param, value| value.nil? }
```

Comment:
Should this be a `.present?` check to match the checks right above it.

Reply:
This statement means that if any search_params has nil value, skips the rest and goes to the next observation resource.

Reply:
I was concerned because `.nil?` isn't the opposite of `.present?` which is used above. But I guess for this check it's ok because any blank value would end up getting turned into nil.

Reply:
Good question. The statement was used before so I didn't change it to avoid any unexpected side effects. 
Logically, `next if search_params.any? { |_param, value| value.nil? }` is the same as `next unless search_params.any? { |_param, value| value.present? }` other than when value==''. Hard decision.

Reply:
But if `value.present?` never returns true, then I think `resolve_element_from_path` will return `nil`, so it should be fine.